src/commands/schemas.ts

```diff
@@ -124,21 +136,152 @@ async function openLatestSchemasCommand(topic: KafkaTopic) {
+
+  // Now loop through draft file:// uris until we find one that doesn't exist.,
+  let chosenFileUri: vscode.Uri | null = null;
+  let draftNumber = -1;
+  while (!chosenFileUri || (await fileUriExists(chosenFileUri))) {
+    draftNumber += 1;
+    const draftFileName = schema.nextVersionDraftFileName(draftNumber);
+    chosenFileUri = vscode.Uri.parse("file://" + `${baseDir}/${draftFileName}`);
+
+    if (draftNumber > 15) {
+      throw new Error(
+        `Could not find a draft file URI that does not exist in the filesystem after 15 tries.`,
+      );
+    }
+  }
+
+  // Now respell to be unknown scheme and add the schema data to the query string,
+  // will become the default schema data for the upload schema command.
+  return vscode.Uri.from({
+    ...chosenFileUri,
+    scheme: "untitled",
+    query: encodeURIComponent(JSON.stringify(schema)),
+  });
+}
+
+/** Check if a file URI exists in the filesystem. */
+async function fileUriExists(uri: vscode.Uri): Promise<boolean> {
+  try {
+    await vscode.workspace.fs.stat(uri);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
 /**
  * Convert a {@link Schema} to a URI and render via the {@link SchemaDocumentProvider} as a read-
  * only document in a new editor tab.
  */
-async function loadOrCreateSchemaViewer(schema: Schema) {
+async function loadOrCreateSchemaViewer(schema: Schema): Promise<vscode.TextEditor> {
   const uri: vscode.Uri = new SchemaDocumentProvider().resourceToUri(schema, schema.fileName());
   const textDoc = await vscode.window.showTextDocument(uri, { preview: false });
-  // VSCode may "throw" an error from `workbench.*.main.js` like `Unknown language: avsc` if the
-  // workspace doesn't have an extension that supports the "avsc" extension/language (or similar).
-  // There isn't anything we can do to suppress those errors (like wrapping the line below in try/catch),
-  // but they don't show up to the user unless they look at the "Window" output channel.
-  vscode.languages.setTextDocumentLanguage(textDoc.document, schema.fileExtension());
+
+  await setEditorLanguageForSchema(textDoc, schema);
+
   return textDoc;
 }
 
+/**
+ * Possibly set the language of the editor's document based on the schema.
+ * Depends on what languages the user has installed.
+ */
+async function setEditorLanguageForSchema(textDoc: vscode.TextEditor, schema: Schema) {
+  const installedLanguages = await vscode.languages.getLanguages();
+
+  for (const language of schema.languageTypes()) {
+    if (installedLanguages.indexOf(language) !== -1) {
+      vscode.languages.setTextDocumentLanguage(textDoc.document, language);
+      logger.info(`Set document language to ${language} for schema ${schema.subject}`);
+      return;
+    } else {
+      logger.warn(`Language ${language} not installed for schema ${schema.subject}`);
+    }
+  }
+
+  logger.warn("Could not find a matching language for schema ${schema.subject}");
```

Comment:
Should we surface this to the user to encourage installing appropriate language extension?

Reply:
Done:

<img width="524" alt="image" src="https://github.com/user-attachments/assets/7daabdb7-525a-4903-98dd-78ef0cad7211">